### PR TITLE
Project movies - Emma

### PR DESCRIPTION
**Pull request by emma-tech**
Link to my deployed project: https://movie-toplist.netlify.app/

My reflections on how this week's project turned out: 
This project was super fun! :) Really happy with it. Wish I had more time to do more cool stuff :) 

Things I'd like to have clarified or explained in more detail:
Still think the useParams hook is a bit tricky to understand..

### CODE REVIEW

File: code/src/components/MovieList.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    /* Using .map() to render the list of movies,
+    mounting MovieThumb-components, showing each movie in a section on the page */
+    return (
+        <section className='movie-list'>
+            {movieList.map((movie) => (
+                 <MovieThumb key={movie.id} {...movie}/>
+            ))}
+        </section>
+    );
+};
```
Comment (stjernberg) [inline code/src/components/MovieList.js:29]: Great comments in the code, which makes it easy to follow what everyting does.

File: code/src/pages/NotFound.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                className='not-found__image'
+                src={Telescope}
+                alt='telescope icon'
+                />
+                <p className='icon-text'>
+                    Icon made by <a href="https://www.flaticon.com/authors/freepik" title="Freepik"> Freepik </a> from <a href="https://www.flaticon.com/" title="Flaticon"> www.flaticon.com </a> 
+                </p>
+        </section>
+    );
+};
```
Comment (stjernberg) [inline code/src/pages/NotFound.js:28]: I like that you make the  "not found page" funny.

File: code/src/styles/moviethumb.css [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        font-size: 20px;
+        margin-bottom: 40px;
+        padding: 10px;
+    }
+
+    .movie-thumb__released {
+        font-size: 12px;
+        padding: 10px;
+    }
+}
```
Comment (stjernberg) [inline code/src/styles/moviethumb.css:85]: Your page looks great with the hover effect with opacity and the popcorn hearder. It doesn't seem like the width on the movie-poster works, if you intended to fit in four posters in a row. On my screen the posters are aligned to the left and there's a blank space to the right.

File: code/src/styles/moviedetails.css [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        margin-top: 40%;       
+    }
+}
+
+@media (max-width: 320px) {
+
+    .movie-details__poster {
+        width: 300px;
+    }
+}
```
Comment (stjernberg) [inline code/src/styles/moviedetails.css:60]: You've done a great job with the mediaqueries in the detailspage. It looks good in all sizes! I also really like the hover effect on the back link.

File: code/src/App.js [hunk 5]
```diff
@@ -1,9 +1,30 @@
...
+          <Route path='/404'>
+            <NotFound />
+          </Route>
+          <Redirect 
+          to='/404'
+          />
+        </Switch>
+      </BrowserRouter>
+  );
+};
```
Comment (stjernberg) [inline code/src/App.js:30]: Nice and clean structure in the App-file.

File: code/src/components/MovieThumb.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                    <p className='movie-thumb__released'> Released: {release_date} </p>
+                    <img 
+                        className='movie-thumb__poster' 
+                        src={`https://image.tmdb.org/t/p/w500/${poster_path}`}
+                        alt={title} 
+                    />
+            </section>
+        </Link>
+    );
+};
```
Comment (stjernberg) [inline code/src/components/MovieThumb.js:23]: Your code has a good stucture the is easy to follow, with clear naming on props and elements. This goes for all your components.

**Review by stjernberg — COMMENTED** [review]
You've done an awesome job Emma with the movies project. 👏  You write really clean code, wich is easy to follow, thanks to good naming on props and elements and dividing the code into smaller components.  I added some comment about the width of the posters, but other than that you're size looks awesome in all screen-sizes. Well done 🤩

File: code/src/components/BackButton.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import 'styles/backbutton.css'
+
+export const BackButton = () => {
+    
+    return (
+        <div>
+            <a className='back-button' href='/'>
```
Comment (EmOlsz) [inline code/src/components/BackButton.js:9]: Or we could use `react-router-dom` package over here :

```
<Link className="back-button" to="/" />
```

File: code/src/pages/NotFound.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+export const NotFound = () => {
+    
+    return (
+        <section className='not-found'>
+            <BackButton />
+                <div className='not-found__message'>
+                    <h1> Ooops... This page doesn't seem to exist. </h1>
+                        <p> Bummer, but there are a lot of other movies to try!</p>
+                </div>
```
Comment (EmOlsz) [inline code/src/pages/NotFound.js:17]: Some wrong indentations in this file

File: code/src/pages/NotFound.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import { BackButton } from 'components/BackButton'
+import Telescope from '../pages/assets/telescope.png'
```
Comment (EmOlsz) [inline code/src/pages/NotFound.js:4]: Uppercase names are reserved for components only, not images ;)

**Pull request closed without merging**